File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/CodeStyleSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.JavaCodeStyleSettings;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.javascript.JavascriptCodeStyleSettings;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableCodeStyleSettings.class)
public interface CodeStyleSettings {

  Optional<JavaCodeStyleSettings> java();

  Optional<JavascriptCodeStyleSettings> javascript();

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/CodeStyleSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle

import com.fasterxml.jackson.annotation.JsonProperty
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.JavaCodeStyleSettings
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.javascript.JavascriptCodeStyleSettings

data class CodeStyleSettings(

    @JsonProperty("Java")
    val java: JavaCodeStyleSettings? = null,

    @JsonProperty("JavaScript")
    val javascript: JavascriptCodeStyleSettings? = null
)
